modules/cache.py

```diff
@@ -73,6 +77,46 @@ def _downsize_cache_to_target_bytes(self, target_bytes:int):
     def clear(self):
         self._downsize_cache_to_target_bytes(0)
 
+    def populateCache(self, cache_file: str):
+        # if the cache file path exist, the file is json file, and it is not empty
+        if os.path.exists(cache_file) and cache_file.endswith(".json") and os.path.getsize(cache_file) > 0:
+            
+            # open the file and read the data
+            with open(cache_file, "r") as f:
+                
+                # json.load converts the json data into python dictionary
+                dict_data = json.load(f)
+            
+            # populate the cache
+            for _, video_info in dict_data.items():
+                self.add(video_info["url"])
```

Comment:
put this in a try/catch, that way if the file is invalid for some reason the server wont crash

Reply:
we still need this